### PR TITLE
SISRP-43016 - Switches filenaming to use feed section list. Adds course name column for crosslisted courses

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-43016

See QA PR #7406

### CODE REVIEW

File: app/models/rosters/csv.rb [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
       @section_id = opts[:section_id] if opts[:section_id].present?
       @enroll_option = opts[:enroll_option] if opts[:enroll_option].present?
       filter_students_by_options
     end
 
+    def is_crosslisted_course?
+      @is_crosslisted_course ||= Proc.new do
+        !!@sections.to_a.find { |sec| !!sec.try(:[], :cross_listing) }
+      end.call
+    end
```
Comment (davidlimcheng) [inline app/models/rosters/csv.rb]: @redconfetti, is there a difference between this and something that might be more readable, like so?
```
@is_crosslisted_course ||= !!@sections.to_a.find { |sec| !! sec.try(:[], :cross_listing) }
```
In other words, what is the `Proc` definition adding?

Comment (redconfetti) [reply app/models/rosters/csv.rb]: @davidlimcheng - You have a point, the Proc isn't needed to ensure that it's only run once.

Comment (redconfetti) [reply app/models/rosters/csv.rb]: @davidlimcheng - It's updated now

**Comment by davidlimcheng** [comment]
Thanks @redconfetti , LGTM, will merge once Travis gives the green light.

**Pull request merged**